packages/components/pages/f-contact-preferences/src/store/_tests/contactPreferences.module.test.js

```diff
@@ -0,0 +1,149 @@
+
+                // Assert
+                expect(commitSpy).toHaveBeenLastCalledWith(UPDATE_PREFERENCES, { preferences: expectedPreferences, preferenceVersionViewed: expectedVersion });
+            });
+        });
+
+        describe('savePreferences ::', () => {
+            it('should call the postPreferences api method with the correct parameters', async () => {
+                // Act
+                await contactPreferencesModule.actions.savePreferences({ state: contactPreferencesViewModel }, { api: httpClientMock, authToken: token });
+
+                // Assert
+                expect(postPreferencesMock).toHaveBeenCalledWith(token, contactPreferencesUpdateModel);
+            });
+        });
+
+        describe('editPreference ::', () => {
+            it(`should call ${UPDATE_PREFERENCE} mutation with the correct data`, () => {
+                // Arrange
+                const commitSpy = jest.fn();
+                const expectedIndex = contactPreferencesViewModel.preferences.findIndex(e => e.key === 'news');
+                const expectedValue = !contactPreferencesViewModel.preferences[expectedIndex].emailValue;
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'news', field: 'emailValue', value: expectedValue });
+
+                // Assert
+                expect(commitSpy).toHaveBeenLastCalledWith(UPDATE_PREFERENCE, { index: expectedIndex, field: 'emailValue', value: expectedValue });
+            });
+
+            it(`should not call ${UPDATE_PREFERENCE} mutation if the key not found`, () => {
+                // Arrange
+                const commitSpy = jest.fn();
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'X-X-X', field: 'emailValue', value: true });
+
+                // Assert
+                expect(commitSpy).not.toHaveBeenCalled();
+            });
+
+            it(`should not call ${UPDATE_PREFERENCE} mutation if the field not found`, () => {
+                // Arrange
+                const commitSpy = jest.fn();
+
+                // Act
+                contactPreferencesModule.actions.editPreference({ state: contactPreferencesViewModel, commit: commitSpy }, { key: 'news', field: 'X-X-X', value: true });
+
+                // Assert
+                expect(commitSpy).not.toHaveBeenCalled();
+            });
+        });
+    });
+
+    describe('mutations ::', () => {
+        describe(`${UPDATE_PREFERENCES} ::`, () => {
+            it('should update preference state with mapped/filtered/sorted preference data', () => {
+                // Arrange
+                const expectedPreferenceVersionViewed = 1;
+                const expectedPreferences = { ...contactPreferencesViewModel }.preferences;
```

Comment:
```suggestion
                const expectedPreferences = contactPreferencesViewModel.preferences;
```

Reply:
Or is this necessary to break a reference?

Reply:
Yes, I need a copy of the model because I need to change it and I don't want it to affect the other tests.